Plutarch/Internal/Ord.hs

```diff
@@ -18,98 +19,96 @@ import Plutarch.Internal.Term (
  )
 import PlutusCore qualified as PLC
 
-{- | Partial ordering relation.
+{- | Total ordering relation.
 
 = Laws
 
-'#<=' must form a partial order. More precisely:
+'#<=' must form a total order. More precisely:
+
+1. @x #<= x@ @=@ @pcon PTrue@ (reflexivity)
+2. @(x #<= y) #&& (y #<= z)@ @=@ @x #<= z@ (transitivity)
+3. @(x #<= y) #|| (y #<= x)@ @=@ @pcon PTrue@ (totality)
+
+Furthermore, '#<' must be an equivalent strict total order to '#<=':
+
+4. @x #< x@ @=@ @pcon PFalse@ (irreflexivity)
+5. @(x #< y) #&& (y #< z)@ @=@ @x #< z@ (transitivity)
+6. @(x #< y) #|| (y #< x) #|| (x #== z)@ @=@ @pcon PTrue@ (trichotomy)
+7. @x #<= y@ @=@ @(x #< y) #|| (x #== y)@ (strict equivalence)
 
-1. @x #<= x@ @=@ @True@ (@#<=@ is reflexive)
-2. @(x #<= y) #&& (y #<= x)@ @=@ @x #== y@ (@#<=@ is anti-symmetric)
-3. @(x #<= y) #&& (y #<= z)@ @=@ @x #<= z@ (@#<= is transitive)
+If you define '#>=', ensure the following holds:
 
-Furthermore, '#<' must be an equivalent strict partial order to '#<=':
+8. @x #>= y@ @=@ @pnot (x #< y)@
 
-4. @x #<= y@ @=@ @(x #< y) #|| (x #== y)@
-5. @x #< x@ @=@ @False@ (@#<@ is irreflexive)
-6. @x #< y@ @=@ @pnot (y #< x)@ (@#<@ is asymmetric)
-7. @(x #< y) #&& (y #< z)@ @=@ @x #< z@ (@#<@ is transitive)
+Lastly, if you define 'pmax' or 'pmin', ensure the following also hold:
 
-Lastly, if you define '#>=' or '#>', ensure that the following also hold:
+9. @pmax # x # y@ @=@ @pmax # y # x@ (commutativity, also for @pmin)
+10. @pmax # x #$ pmax y z@ @=@ @pmax # (pmax # x # y) # z@ (associativity,
+    also for @pmin)
+11. @pmax # x #$ pmin # y # z@ @=@ @pmin # (pmax # x # y) # (pmax # x # z)@
+    ('pmax' distributes over 'pmin', also equivalent for 'pmin')
+12. @pmin x y@ @=@ @pif' (x #<= y) x y@
+13. @pmax x y@ @=@ @pif' (x #<= y) y x@
 
-8. @x #> y@ @=@ @y #< x@
-9. @x #>= y@ @=@ @pnot (x #< y)@
+Laws 8-13 hold if you use the defaults provided by this type class.
 
-The default implementations of '#>=' and '#>' ensure these laws.
+@since WIP
 -}
-class PEq t => PPartialOrd t where
+class PEq t => POrd t where
+  -- | @since WIP
+  {-# INLINEABLE (#<=) #-}
   (#<=) :: Term s t -> Term s t -> Term s PBool
   default (#<=) :: POrd (PInner t) => Term s t -> Term s t -> Term s PBool
   x #<= y = pto x #<= pto y
+
+  -- | @since WIP
+  {-# INLINEABLE (#<) #-}
   (#<) :: Term s t -> Term s t -> Term s PBool
   default (#<) :: POrd (PInner t) => Term s t -> Term s t -> Term s PBool
   x #< y = pto x #< pto y
 
   -- | @since WIP
+  {-# INLINEABLE (#>=) #-}
   (#>=) :: forall (s :: S). Term s t -> Term s t -> Term s PBool
   x #>= y = pnot #$ x #< y
```

Comment:
Can't this also be argument re arrangement of #<= ?

Reply:
Hmm, yeah, good point.